source/common/quic/envoy_quic_client_connection.h

```diff
@@ -3,13 +3,28 @@
 #include "envoy/event/dispatcher.h"
 
 #include "common/network/utility.h"
-#include "common/quic/envoy_quic_connection.h"
+#include "common/quic/envoy_quic_utils.h"
+#include "common/quic/quic_network_connection.h"
+
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#pragma GCC diagnostic ignored "-Winvalid-offsetof"
```

Comment:
Having to duplicate these `__GNUC__` diagnostic ignores in 4 places doesn't seem ideal.